### ISSUE TITLE
Suspense

### ISSUE THREAD

**Issue by will-hart** [facebook/react#14188]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

I have a component which listens for resize events (via a BlueprintJS ResizeSensor). When loading a component dynamically with `lazy` / `Suspense`, an exception occurs as the resize sensor appears to be unmounted:

```
Uncaught Error: Unable to find node on an unmounted component.
    at invariant (29.chunk.js:86295)
    at findCurrentFiberUsingSlowPath (29.chunk.js:90628)
    at findCurrentHostFiber (29.chunk.js:90640)
    at findHostInstanceWithWarning (29.chunk.js:106349)
    at findDOMNode (29.chunk.js:106869)
    at ResizeSensor.componentDidUpdate (29.chunk.js:10535)
```

The resize sensor [should be removing listeners on unmount](https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/resize-sensor/resizeSensor.tsx#L73).

Demo: https://codesandbox.io/s/n4241q075l
Related: https://github.com/palantir/blueprint/issues/3141

**What is the expected behavior?**

No exception is thrown.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

Tested with:
- React 16.6.1, fails
- React 16.6.0, works

**Comment by arianon** [comment]
Nope, it's broken on master, and after bisecting, I confirmed the regression was caused by #14083

**Comment by gaearon** [comment]
>EDIT: Something else I found is that before the regression, ResizeSensor.componentDidUpdate doesn't execute, but now it does.

This sounds like a separate problem, can you provide a test case for it?

**Comment by gaearon** [comment]
Here's a minimal repro with ReactDOM:

```js
const Lazy = React.lazy(() => new Promise(resolve => resolve({
  default() {
    return 'hello';
  }
})))

class Child extends React.Component {
  componentDidUpdate() {
    ReactDOM.findDOMNode(this);
  }

  render() {
    return null;
  }
}

class App extends React.Component {
  state = {
    suspend: false
  };
  handleClick = () => {
    this.setState({ suspend: true });
  };
  render() {
    return (
      <React.Suspense fallback="Loading">
        <Child />
        <button onClick={this.handleClick}>Suspend</button>
        <Child />
        {this.state.suspend && <Lazy />}
      </React.Suspense>
    );
  }
}

ReactDOM.render(
  <App />,
  document.getElementById('container')
);
```

**Comment by gaearon** [comment]
Okay, so a part of this is covered by https://github.com/facebook/react/pull/14197. It got first broken by #14083 and got fixed in #14164.

However I think repro case in https://github.com/facebook/react/issues/14188#issuecomment-437705392 might be a separate issue. Looking into it.

**Comment by arianon** [comment]
>This sounds like a separate problem, can you provide a test case for it?

Welp, turns out I'm wrong, I actually made a mistake when mocking `componentDidUpdate` which resulted in it not being called :stuck_out_tongue_closed_eyes:

**Comment by gaearon** [comment]
This was likely fixed in https://github.com/facebook/react/pull/15312. Can you verify on master?

**Comment by zombieJ** [comment]
@gaearon @acdlite,
Seems it still happens. I tried to get mini reproduce but it seems happened in complex environment.

User report reproduce: https://codesandbox.io/s/antd-reproduction-template-sbsiz
Quick click the nav item will throw with `Unable to find node on an unmounted component.`

ref: https://github.com/ant-design/ant-design/issues/17016

**Comment by gaearon** [comment]
Sorry, we didn’t release the fix yet.

**Comment by joshnabbott** [comment]
@gaearon any ETA on when this will be available?

**Comment by ambroselittle** [comment]
We are blocked by this, too (without a pretty significant workaround). Any sense of a release timeframe? @gaearon? Is there some way to find out (in general) when the "next" release is? Sorry if this is covered somewhere.. 

For the record, this is surfacing for us through Semantic UI React Dropdown. So we're beholden to them as well, but they point us here for the fix. :)

**Comment by wenisy** [comment]
any new updates?

**Comment by wenisy** [comment]
> Sorry, we didn’t release the fix yet.

@gaearon 

any new updates?

**Comment by gaearon** [comment]
We've published `0.0.0-db3ae32b8` of all packages, which includes this fix. It is likely to be a release candidate for `16.9.0`. You can give it a try now and see whether it resolves the issue for you. Thanks.

**Comment by gaearon** [comment]
(You should still ask packages to migrate away from `findDOMNode` though as it's eventually going to get deprecated.)

**Comment by MissLlq** [comment]
> We've published `0.0.0-db3ae32b8` of all packages, which includes this fix. It is likely to be a release candidate for `16.9.0`. You can give it a try now and see whether it resolves the issue for you. Thanks.

16.9.0 hadn't  fix this bug!

**Comment by eps1lon** [comment]
> > We've published `0.0.0-db3ae32b8` of all packages, which includes this fix. It is likely to be a release candidate for `16.9.0`. You can give it a try now and see whether it resolves the issue for you. Thanks.
> 
> 16.9.0 hadn't fix this bug!

@MissLlq  All reproductions linked in this issue (https://codesandbox.io/s/zwzqp18ool and https://codesandbox.io/s/n4241q075l) no longer crash with 16.9.

If you still encounter this error message with 16.9 please open a new issue.
